avm/ptn/aca-lza/hosting-environment/modules/application-gateway/deploy.app-gateway.bicep

```diff
@@ -0,0 +1,394 @@
+param location string = resourceGroup().location
+
+@description('Optional. The tags to be assigned to the created resources.')
+param tags object = {}
+
+@description('The FQDN of the Application Gateawy. Must match the TLS certificate.')
+param applicationGatewayFqdn string
+
+@description('The existing subnet resource ID to use for Application Gateway.')
+param applicationGatewaySubnetId string
+
+@description('The FQDN of the primary backend endpoint.')
+param applicationGatewayPrimaryBackendEndFqdn string
+
+@description('The path to use for Application Gateway\'s backend health probe.')
+param appGatewayBackendHealthProbePath string = '/'
+
+@description('The base64 encoded certificate to use for Application Gateway certificate. If this is provided, the certificate will be added to the Key Vault.')
+param base64Certificate string = ''
+
+@description('The name of the certificate key to use for Application Gateway certificate.')
+param applicationGatewayCertificateKeyName string
+
+@description('The resource ID of the exsiting Log Analytics workload for diagnostic settngs, or nothing if you don\'t need any.')
+param applicationGatewayLogAnalyticsId string = ''
+
+@description('The resource ID of the existing Key Vault which contains Application Gateway\'s cert.')
+param keyVaultId string
+
+@description('Optional, default value is true. If true, any resources that support AZ will be deployed in all three AZ. However if the selected region is not supporting AZ, this parameter needs to be set to false.')
+param deployZoneRedundantResources bool = true
+
+@description('Optional. DDoS protection mode for the Public IP of the Application Gateway. See https://learn.microsoft.com/azure/ddos-protection/ddos-protection-sku-comparison#skus')
+param enableDdosProtection bool = false
+
+// ------------------
+// VARIABLES
+// ------------------
+
+var keyVaultIdTokens = split(keyVaultId, '/')
+
+@description('The subscription ID of the existing Key Vault.')
+var keyVaultSubscriptionId = keyVaultIdTokens[2]
+
+@description('The name of the resource group containing the existing Key Vault.')
+var keyVaultResourceGroupName = keyVaultIdTokens[4]
+
+@description('The name of the existing Key Vault.')
+var keyVaultName = keyVaultIdTokens[8]
+
+// ------------------
+// RESOURCES
+// ------------------
+
+@description('User-configured naming rules')
+module naming '../naming/naming.module.bicep' = {
+  name: take('agwNamingDeployment-${deployment().name}', 64)
+  params: {
+    uniqueId: uniqueString(resourceGroup().id)
+    environment: environment
+    workloadName: workloadName
+    location: location
+  }
+}
+
+// TODO: Check if this is required if enableApplicationCertificate is false
+@description('A user-assigned managed identity that enables Application Gateway to access Key Vault for its TLS certs.')
+module userAssignedIdentity 'br/public:avm/res/managed-identity/user-assigned-identity:0.2.1' = {
```

Comment:
For all references to AVM modules please ensure you pass in `enableTelemetry: enableTelemetry` (you may need to add parameters for it) all the way from the top `main.bicep`. Only then a user can effectively disable telemetry if they so choose.

For example 
```bicep
module applicationGatewayPublicIp 'br/public:avm/res/network/public-ip-address:0.4.1' = {
  name: take('applicationGatewayPublicIp-Deployment-${uniqueString(resourceGroup().id)}', 64)
  params: {
    location: location
    name: naming.outputs.resourcesNames.applicationGatewayPip
    enableTelemetry: enableTelemetry
    (...)
    }
}
```